addons/account/models/account_move.py

```diff
@@ -3437,34 +3436,56 @@ def _message_post_after_hook(self, new_message, message_values):
+    def _creation_subtype(self):
+        # OVERRIDE
+        if self.move_type in ('out_invoice', 'out_refund', 'out_receipt'):
+            return self.env.ref('account.mt_invoice_created')
+        else:
+            return super(AccountMove, self)._creation_subtype()
 
-        :returns:   A list of tuples (priority, method) where method takes an attachment as parameter.
-        """
-        return []
+    def _track_subtype(self, init_values):
+        # OVERRIDE to add custom subtype depending of the state.
+        self.ensure_one()
 
-    def _get_update_invoice_from_attachment_decoders(self, invoice):
-        """ Returns a list of method that are able to create an invoice from an attachment and a priority.
+        if not self.is_invoice(include_receipts=True):
+            if self.payment_id and 'state' in init_values:
+                self.payment_id._message_track(['state'], {self.payment_id.id: init_values})
+            return super(AccountMove, self)._track_subtype(init_values)
 
-        :param invoice: The invoice on which to update the data.
-        :returns:       A list of tuples (priority, method) where method takes an attachment as parameter.
-        """
-        return []
+        if 'payment_state' in init_values and self.payment_state == 'paid':
+            return self.env.ref('account.mt_invoice_paid')
+        elif 'state' in init_values and self.state == 'posted' and self.is_sale_document(include_receipts=True):
+            return self.env.ref('account.mt_invoice_validated')
+        return super(AccountMove, self)._track_subtype(init_values)
 
-    @api.depends('move_type', 'partner_id', 'company_id')
-    def _compute_narration(self):
-        use_invoice_terms = self.env['ir.config_parameter'].sudo().get_param('account.use_invoice_terms')
-        for move in self.filtered(lambda am: not am.narration):
-            if not use_invoice_terms or not move.is_sale_document(include_receipts=True):
-                move.narration = False
-            else:
-                if not move.company_id.terms_type == 'html':
-                    narration = move.company_id.invoice_terms if not is_html_empty(move.company_id.invoice_terms) else ''
-                else:
-                    baseurl = self.env.company.get_base_url() + '/terms'
-                    narration = _('Terms & Conditions: %s', baseurl)
-                move.narration = narration or False
+    def _creation_message(self):
+        # OVERRIDE
+        if not self.is_invoice(include_receipts=True):
+            return super()._creation_message()
+        return {
+            'out_invoice': _('Invoice Created'),
+            'out_refund': _('Credit Note Created'),
+            'in_invoice': _('Vendor Bill Created'),
+            'in_refund': _('Refund Created'),
+            'out_receipt': _('Sales Receipt Created'),
+            'in_receipt': _('Purchase Receipt Created'),
+        }[self.move_type]
```

Comment:
could use `_lt` here (and probably `str()`), this way we make only one lookup in the translations instead of 6